Tests/OAuth/ResourceOwner/AzureResourceOwner.php

```diff
@@ -0,0 +1,101 @@
+{
+    protected $csrf = true;
+
+    protected $userResponse = <<<json
+{
+    "id":  "1",
+    "foo": "bar"
+}
+json;
+
+    protected $paths = array(
+        'identifier'     => 'sub',
+        'nickname'       => 'unique_name',
+        'realname'       => array('given_name', 'family_name'),
+        'email'          => array('upn', 'email'),
+        'profilepicture' => null,
+    );
+
+    protected $expectedUrls = array(
+        'authorization_url_csrf' => 'http://user.auth/?test=2&response_type=code&client_id=clientid&state=random&redirect_uri=http%3A%2F%2Fredirect.to%2F&resource=https%3A%2F%2Fgraph.windows.net',
+    );
+
+    public function testGetAuthorizationUrl()
+    {
+        $this->assertEquals(
+            $this->options['authorization_url'] .'&response_type=code&client_id=clientid&state=random&redirect_uri=http%3A%2F%2Fredirect.to%2F&resource=https%3A%2F%2Fgraph.windows.net',
+            $this->resourceOwner->getAuthorizationUrl('http://redirect.to/')
+        );
+    }
+
+    public function testGetUserInformation()
+    {
+        $token = base64_encode($this->userResponse) . '.';
+
+        /**
+         * @var $userResponse \HWI\Bundle\OAuthBundle\OAuth\Response\AbstractUserResponse
+         */
+        $userResponse = $this->resourceOwner->getUserInformation(array(
+            'access_token' => 'token',
+            'id_token' =>  $token
+        ));
+
+        // can't really test real responses because of the weird id_token value; seems strangely base64 encoded
+        $this->assertNull($userResponse->getRefreshToken());
+        $this->assertNull($userResponse->getExpiresIn());
+    }
+
+    public function testCustomResponseClass()
+    {
+        $this->markTestSkipped('Can\' test custom response because of the way the id_token value is set; is always returning null');
+
+        $class         = '\HWI\Bundle\OAuthBundle\Tests\Fixtures\CustomUserResponse';
+        $resourceOwner = $this->createResourceOwner($this->resourceOwnerName, array('user_response_class' => $class));
+
+        $this->mockBuzz();
+
+        $token = base64_encode($this->userResponse) . '.';
+
+        /**
+         * @var $userResponse \HWI\Bundle\OAuthBundle\OAuth\Response\AbstractUserResponse
+         */
+        $userResponse = $this->resourceOwner->getUserInformation(array(
+            'access_token' => 'token',
+            'id_token' =>  $token
+        ));
+
+        $this->assertInstanceOf($class, $userResponse);
+        $this->assertEquals('foo666', $userResponse->getUsername());
+        $this->assertEquals('foo', $userResponse->getNickname());
+        $this->assertEquals('token', $userResponse->getAccessToken());
+        $this->assertNull($userResponse->getRefreshToken());
+        $this->assertNull($userResponse->getExpiresIn());
+    }
+
+    protected function setUpResourceOwner($name, $httpUtils, array $options)
+    {
+        $options = array_merge(
```

Comment:
Protected `$options` property should be overridden instead.

Can you confirm @stloyd?